rust/src/io/object_store.rs

```diff
@@ -49,6 +48,20 @@ pub struct ObjectStore {
     prefetch_size: usize,
 }
 
+/// URI schemes supported by lance
+mod schemes {
+    static S3: &str = "s3";
+    static GOOGLE_CLOUD: &str = "gs";
```

Comment:
Could we rename these variable to be more consistent?
i.e., either `S3/GCS`, or `AWS_S3 / GOOGLE_GCS`. 

Also not sure we want to different "Local file systme" and "remote file system" tho.